### PR TITLE
[api] Add NetworkStatus to allocation response

### PR DESCRIPTION
**Pull request by deverton-godaddy**
Service discovery or mesh network systems consuming the Nomad event stream or API need to know the CNI assigned IP for the allocation. This data is returned by the underlying Nomad API but isn't mapped in the response struct in the Go client.

### CODE REVIEW

**Review by lgfa29 — APPROVED** [review]
Thank you for the contribution @deverton-godaddy!

I pushed a commit to add a CHANGELOG entry and a quick test. I also reordered the struct fields so it matches the order in `structs.go`, making it easier to finding missing fields (and I did notice some others missing here 😅)

Before we can merge this we will need you to sign the CLA. Would you be able to do that?

Thanks!

**Comment by deverton-godaddy** [comment]
Thank you for taking the time to update this PR. I'm still working on getting CLA approval and will update as soon as I can. I'm hoping it should be in the next few days.

**Comment by deverton-godaddy** [comment]
@lgfa29 I've signed the CLA so this should be good to go.

**Comment by lgfa29** [comment]
> @lgfa29 I've signed the CLA so this should be good to go.

Ops, sorry I missed this @deverton-godaddy. Thanks for handling the CLA!

**Pull request merged**
